src/Utils/Helper.php

```diff
@@ -0,0 +1,282 @@
+     */
+    public static function extract_title($data)
+    {
+        if (!array_key_exists('context', $data) || !is_array($data['context'])) {
+            return null;
+        }
+
+        if (!array_key_exists('custom', $data['context']) || !is_array($data['context']['custom'])) {
+            return null;
+        }
+
+        if (array_key_exists('caption', $data['context']['custom'])) {
+            return $data['context']['custom']['caption'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public static function extract_description($data)
+    {
+        if (!array_key_exists('context', $data) || !is_array($data['context'])) {
+            return null;
+        }
+
+        if (!array_key_exists('custom', $data['context']) || !is_array($data['context']['custom'])) {
+            return null;
+        }
+
+        if (array_key_exists('alt', $data['context']['custom'])) {
+            return $data['context']['custom']['alt'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public static function extract_credit($data)
+    {
+        if (!array_key_exists('image_metadata', $data)) {
+            return null;
+        }
+
+        $metadata = $data['image_metadata'];
+
+        if (array_key_exists('Copyright', $metadata)) {
+            return $metadata['Copyright'];
+        }
+
+        if (array_key_exists('By-line', $metadata)) {
+            return $metadata['By-line'];
+        }
+
+        if (array_key_exists('Artist', $metadata)) {
+            return $metadata['Artist'];
+        }
+
+        if (array_key_exists('Creator', $metadata)) {
+            return $metadata['Creator'];
+        }
+
+        if (array_key_exists('XPAuthor', $metadata)) {
+            return $metadata['XPAuthor'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public static function determine_format($data)
+    {
+        $url = $data['secure_url'];
+
+        if ($extension = pathinfo($url, PATHINFO_EXTENSION)) {
+            return $extension;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $data
+     * @param int $status
+     * @param int $ttl
+     * @return HTTPResponse
+     */
+    public static function json($data, $status = 200, $ttl = 300)
+    {
+        $response = new HTTPResponse();
+
+        $response->setBody(json_encode($data));
+
+        $response->setStatusCode($status);
+
+        $expire = new DateTime();
+        $expire->add(new DateInterval('PT' . $ttl . 'S'));
+        $expire->setTimezone(new DateTimeZone('UTC'));
+
+        $response->addHeader('Content-Type', 'application/json; charset=utf-8')
+            ->addHeader('Access-Control-Allow-Methods', 'GET')
+            ->addHeader('Access-Control-Allow-Credentials', 'true')
+            ->addHeader('Access-Control-Allow-Origin', '*')
```

Comment:
As you're using `Vary: Origin` then should pass the request as a parameter and use the Origin in the response (can also add some whitelisting)

Reply:
Not doing that, request is not available/needed. This is only internal.

Reply:
Potentially also a callback for Cloudinary